### ISSUE TITLE
[python-ray] depends on specific version (<3.5.0) of redis

### ISSUE THREAD

**Issue by huyz-git** [acxz/pkgbuilds#80]
```
$ ray

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 567, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 884, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 775, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (redis 3.5.3 (/usr/lib/python3.8/site-packages), Requirement.parse('redis<3.5.0,>=3.3.2'), {'ray'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/ray", line 33, in <module>
    sys.exit(load_entry_point('ray==1.0.0', 'console_scripts', 'ray')())
  File "/usr/bin/ray", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/lib/python3.8/site-packages/ray/__init__.py", line 56, in <module>
    import ray._raylet  # noqa: E402
  File "python/ray/_raylet.pyx", line 102, in init ray._raylet
  File "/usr/lib/python3.8/site-packages/ray/exceptions.py", line 8, in <module>
    from ray.core.generated.common_pb2 import RayException, Language
  File "/usr/lib/python3.8/site-packages/ray/core/generated/common_pb2.py", line 7, in <module>
    from google.protobuf.internal import enum_type_wrapper
  File "/usr/lib/python3.8/site-packages/google/protobuf/__init__.py", line 37, in <module>
    __import__('pkg_resources').declare_namespace(__name__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3239, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3222, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3251, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 569, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 582, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 770, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'redis<3.5.0,>=3.3.2' distribution was not found and is required by ray
```

While the `python-redis` package in community repo is `3.5.3`
```
$ pacman -Ss python-redis                                     
                     
community/python-redis 3.5.3-1
    The Python interface to the Redis key-value store
```

**Comment by acxz** [comment]
I don't really want to micro manage the python deps. I think the best solution would be to make a PR upstream on relaxing the `redis` dep.

**Comment by acxz** [comment]
I made an upstream PR resolving this error: https://github.com/ray-project/ray/pull/11776
In the meanwhile if you want to make a patch for it, feel free to make a PR and I'll merge and push it out on the AUR.

**Comment by acxz** [comment]
patched with 36c77d9

**Comment by acxz** [comment]
closed with https://github.com/acxz/pkgbuilds/commit/83845629ea3df05ad0c525e6a503988e6250df18
